Build sdist inside of a container

shanemcd opened the pull request:
A lot of people have experienced issues with the system-level dependencies that are required in order to build the source distribution that is handed off to the image builds. This makes it unnecessary to install any additional software on the host machine aside from Ansible and Docker.

I've successfully tested this with native Docker on Fedora 26 and Docker For Mac.

While this eliminates the need for installing any additional tooling, there are some downsides:

- While native Docker on Linux does not incur the overhead of copying files to / from a VM,  I/O with Docker For Mac is ***slow***. This substantially increases the amount of time it takes to create the source distribution on macOS.
- We incur the cost of building this image. On a fast connection and decent hardware the time added is negligible, but it is something worth noting.

I personally think these tradeoffs are worth the benefits, but I'll let others weigh in as to whether or not we want to go this route.

Link #83

shanemcd committed:
Build sdist inside of a container

A lot of people have experienced issues with the system-level dependencies that are required in order to build the source distribution that is handed off to the image builds. This makes it unnecessary to install any additional software on the host machine aside from Ansible and Docker.

matburt reviewed (approved):
I was on the fence about this but given the amount of support requests we are getting I think I'm fine with it now.

matburt reviewed (changes requested):
Actually, I'm wondering if you can update the INSTALL docs to reflect this behavior change?

shanemcd committed:
Update INSTALL.md

These dependencies are unnecessary now that we are building the sdist inside of a container.

shanemcd commented:
@martburt Done.

shanemcd commented:
@Numblesix The image created by this PR should not need any information about the AWX environment, it simply builds the source distribution which is handed off to later image builds. If you are experiencing a problem please open a new issue with some details.
